### PR TITLE
fix: config to optimize build and README instruction

### PR DESCRIPTION
**Pull request by austinabell**
I don't have the bandwidth to fix the contract compilation from #15/#16, but figured I'd PR in some other quick config things to optimize the build (contract will be a small fraction of the size) and add a missing install instruction (as it was unclear to me) on the way to running into the same issue.

Hopefully this is helpful!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,11 @@ Launch the web service:
 cargo run --bin battleship-web-server --release
 ```
 
+Install trunk:
+```
+cargo install trunk
+```
+
```
Comment (flaub) [inline README.md:53]: There's a link above for installing Yew: 
https://yew.rs/docs/getting-started/introduction

File: contract/.cargo/config.toml [hunk 2]
```diff
@@ -1,2 +1,5 @@
 [build]
 target = "wasm32-unknown-unknown"
+
+[target.wasm32-unknown-unknown]
+rustflags = ["-C", "link-arg=-s"]
```
Comment (flaub) [inline contract/.cargo/config.toml:5]: What's the intent with this?

File: contract/Cargo.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@ version = "0.1.0"
 edition = "2021"
 
 [lib]
-crate-type = ["cdylib", "rlib"]
+crate-type = ["cdylib"]
```
Comment (flaub) [inline contract/Cargo.toml:7]: I believe this was following the NEAR docs. What's the intent with this change?

File: contract/.cargo/config.toml (hunk 2 above)
Comment (austinabell) [reply contract/.cargo/config.toml:5]: This just minimizes the amount of bloat within the built wasm file

**Comment by austinabell** [comment]
Closing because stale and example is deprecated

**Pull request closed without merging**